src/test/CalculatorPluginTest.vala

```diff
@@ -21,105 +21,131 @@ class Synapse.CalculatorPluginTest : Object {
+        assert_equal ("sqrt(-1 + 423 + 1) + (3.23 * 8.56) - s(90 + 0.2)", "47.428606036");
 
         // Modulos and percentages
-        assert_throw ("723 mod 5", 3); // 'bc' does not handle 'mod'
-        assert_throw ("2%", 0.02); // 'bc' treats '%' as 'mod'
-        assert_throw ("(2.0 + 2.0)% -3.0", 1.0); //equiv "4 - (4 / 3 *3)" for doubles (essentially 0)
-        assert_equal ("scale=0;(2 + 2)% - 3", 1); //equiv "4mod(-3)" for integers
-        assert_throw ("10 + 5 - 10%", 14.9); // https://github.com/elementary/calculator/issues/44
-        assert_equal ("10 - 10/100 + 5", 14.9); // https://github.com/elementary/calculator/issues/44
+        assert_throw ("723 mod 5", "3"); // 'bc' does not handle 'mod'
+        assert_throw ("2%", "0.02"); // 'bc' treats '%' as 'mod'
+        assert_throw ("(2.0 + 2.0)% -3.0", "1.0"); //equiv "4 - (4 / 3 *3)" for doubles (essentially 0)
+        assert_equal ("scale=0;(2 + 2)% - 3", "1"); //equiv "4mod(-3)" for integers
+        assert_throw ("10 + 5 - 10%", "14.9"); // https://github.com/elementary/calculator/issues/44
+        assert_equal ("10 - 10/100 + 5", "14.9"); // https://github.com/elementary/calculator/issues/44
 
         // Math constants
-        assert_throw ("pi", 3.141592654);
-        assert_throw ("pi - 2", 1.141592654); // https://github.com/elementary/calculator/issues/59
-        assert_throw ("(π)", 3.141592654);
-        assert_throw ("e", 2.718281828);
+        assert_equal ("pi", "3.141592654"); // Constant name converted
+        assert_equal ("pi - 2", "1.141592654");
+        assert_equal ("(π)", "3.141592654"); // Constant name converted
+        assert_throw ("e", "2.718281828");
 
         // Trigonometry
-        assert_throw ("sin(pi / 2)", 1);
-        assert_throw ("sin(-pi)", 0); // https://github.com/elementary/calculator/issues/1
-        assert_throw ("cos(90)", -0.448073616);
-        assert_throw ("sinh(2)", 3.626860408);
-        assert_throw ("cosh(2)", 3.762195691);
-        assert_equal ("s(0)", 0); // Equiv "sin(0)"
-        assert_equal ("c(0)", 1); // Equiv "cos(0)"
-        assert_equal ("a(1)*4", 3.14159265358979323844); // Equiv "arctangent (1 radian) * 4" which equals pi
-        assert_equal ("pi=a(1)*4; c(pi)", -1); // Equiv "cosine (pi / 2)"
-        assert_equal ("pi=a(1)*4; s(pi / 2)", 1); // Equiv "sine (pi / 2)"
-        assert_throw ("sinh(2)", 3.626860408);
-        assert_throw ("cosh(2)", 3.762195691);
-        assert_equal ("s(0.123)^2 + c(0.123)^2", 1); //Equiv "sin^2(x) + cos^2(x) = 1"
-        assert_throw ("sin(0.123)^2 + cos(0.123)^2", 1); //function names not recognised by `bc`
-        assert_throw ("tan(0.245) - sin(0.245) / cos(0.245)", 0);//function names not recognised by `bc`
-        assert_throw ("asin(0.532) + acos(0.532)", 1.570796327);//function names not recognised by `bc`
-        assert_throw ("atan(1)", 0.785398163);
+        assert_equal ("sin(pi / 2)", "1");
```

Comment:
While it didn't work at all before now `sin(pi / 2)` responds with `1.00000000000000000000`, any idea why these 0's aren't omitted? Maybe I should open a separate issue for it?

Reply:
That is what is returned by bc (with the default scale).  We do not do any rounding within the plugin.